### PR TITLE
fix: set the runners for github action jobs dynamically

### PR DESCRIPTION
**Pull request by bas-info-nl**
Runners can't be set through environment variables, but can be set via a job output.

new line for build

### CODE REVIEW

**Review by edgarvonk — APPROVED** [review]

File: .github/workflows/build-test-deploy.yml [hunk 1]
```diff
@@ -29,8 +28,19 @@ permissions:
   pull-requests: write
 
 jobs:
+  set-runner:
```
Comment (edgarvonk) [inline .github/workflows/build-test-deploy.yml:31]: What issue does this solve? Seems to me this makes our workflow unnecessarily more complex?

I also don't see this back in the GitHub documentation. Why not leave it as it currently is?

Comment (bas-info-nl) [reply .github/workflows/build-test-deploy.yml:31]: zie bv https://stackoverflow.com/questions/71961921/specify-runner-to-be-used-depending-on-condition-in-a-github-actions-workflow

De env var op de runner werkt niet blijkt (vreemd dat ik dat niet eerder heb gezien) en
anders kan ik het beter weer weg halen. Dan zetten we de runner op elke job weer terug naar ubuntu-latest. Dit is ook niet zo'n groot ding, het is enkel om 1 explicite versie van de runner te kunnen zetten...

Comment (edgarvonk) [reply .github/workflows/build-test-deploy.yml:31]: I mean: I would just go back to `runs-on: ubuntu-latest` like we had really. Not such a big deal. Or `ubuntu-22.04` if we wish to be explicit. Dependabot will make PRs to update these anyway. :-)

**Review by edgarvonk — APPROVED** [review]
ok, I'll approve it. But I do think it's a bit overkill really.

File: .github/workflows/build-test-deploy.yml (hunk 1 above)
Comment (bas-info-nl) [reply .github/workflows/build-test-deploy.yml:31]: Yes, let's do that - this does make it more complex than needed

**Review by Jorann — APPROVED** [review]
Very overkill,

But i need it to work on main

**Pull request merged**